Unable to compile catboostmodel.dll via .NET

catboost version: Latest
Operating System: Windows 10 Professional
CPU: AMD Ryzen 16 core
Python 3.7

I'm currently having trouble building the dll via .NET.  I tried to build the solution but was getting the following error message:

>     File "C:\Users\me\Downloads\catboost-master\catboost-master\build\ymake_conf.py", line 1686, in __init__
>   
>       self.sdk_version = os.environ.get('WindowsSDKVersion').replace('\\', '')
>   
>   AttributeError: 'NoneType' object has no attribute 'replace'

It seems I didn't have a few things in my environment variables: WindowsSDKDir, VCInstallDir and WindowsSDKVersion.  I did some googling and found what I thought was the answer here:

https://github.com/catboost/catboost/issues/393#issuecomment-412428144

I ran the vcvars bat file and was able to see that these environment variables were there by executing a few echo statements, however I got the same error.  Then what I did was manually add these variables as environment variables in Windows.  Ran the solution and it failed to build again but this time I got a different error.

>   Traceback (most recent call last):
>     File "devtools/ya/app.py", line 350, in configure_exit_interceptor
>       yield
>     File "devtools/ya/app.py", line 56, in helper
>       return action(args)
>     File "devtools/ya/entry/entry.py", line 45, in do_main
>       res = handler.handle(handler, args, prefix=['ya'])
>     File "devtools/ya/core/handler.py", line 157, in handle
>       return handler.handle(self, args[1:], prefix + [name])
>     File "devtools/ya/core/dispatch.py", line 37, in handle
>       return self.command().handle(root_handler, args, prefix)
>     File "devtools/ya/core/handler.py", line 337, in handle
>       return self._action(params)
>     File "devtools/ya/app.py", line 78, in helper
>       return action(ctx.params)
>     File "devtools/ya/build/build_handler.py", line 71, in do_ya_make
>       builder = ya_make.YaMake(params, app_ctx)
>     File "devtools/ya/build/ya_make.py", line 635, in __init__
>       self.ctx = Context(self.opts, app_ctx=app_ctx, graph=graph, tests=tests, configure_errors=configure_errors, make_files=make_files, lite_graph=lite_graph)
>     File "devtools/ya/build/ya_make.py", line 510, in __init__
>       self.output_result = result_store.ResultStore(self.output_dir) if self.create_output else None
>     File "devtools/ya/yalibrary/runner/result_store.py", line 13, in __init__
>       exts.fs.create_dirs(self.root)
>     File "library/python/windows/__init__.py", line 81, in f_wrapped
>       reraise(tp, value, tb)
>     File "library/python/windows/__init__.py", line 77, in f_wrapped
>       return f(*args, **kwargs)
>     File "library/python/fs/__init__.py", line 55, in create_dirs
>       ensure_dir(path)
>     File "library/python/windows/__init__.py", line 81, in f_wrapped
>       reraise(tp, value, tb)
>     File "library/python/windows/__init__.py", line 77, in f_wrapped
>       return f(*args, **kwargs)
>     File "library/python/fs/__init__.py", line 43, in ensure_dir
>       os.makedirs(path)
>     File "contrib/tools/python/src/Lib/os.py", line 157, in makedirs
>   WindowsError: [Error 123] The filename, directory name, or volume label syntax is incorrect: 'C:\\Users\\me\\Downloads\\catboost-master\\catboost-master\\catboost\\dotnet\\Debug" --verbose'
> C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\Common7\IDE\VC\VCTargets\Microsoft.MakeFile.Targets(44,5): error MSB3073: The command ""C:\Users\me\Downloads\catboost-master\catboost-master\catboost\dotnet\../../ya" make "C:\Users\me\Downloads\catboost-master\catboost-master\catboost\dotnet\../../catboost/libs/model_interface" --output "C:\Users\me\Downloads\catboost-master\catboost-master\catboost\dotnet\Debug\" --verbose" exited with code 1.

I tried this on my laptop, desktop and even work computer.  It's not clear why I'd get this file/directory/syntax error.  The directory and path looks correct to me.  Also, is there a bug in the logic relating to these environment variables. They don't seem to get picked up unless I manually change environment variables which I'm sure is not the intention.  

Due to speed of inference we are looking to utilise catboost vs the other boosting algorithms.

--- a/catboost/dotnet/HeartDiseaseDemo/HeartDiseaseDemo.csproj
+++ b/catboost/dotnet/HeartDiseaseDemo/HeartDiseaseDemo.csproj
@@ -1,7 +1,7 @@
 <Project Sdk="Microsoft.NET.Sdk.Web">
 
   <PropertyGroup>
-    <TargetFramework>netcoreapp2.2</TargetFramework>
+    <TargetFramework>netcoreapp2.1</TargetFramework>
     <AspNetCoreHostingModel>InProcess</AspNetCoreHostingModel>
   </PropertyGroup>
 
@@ -11,9 +11,12 @@
 
 
   <ItemGroup>
-    <PackageReference Include="CatBoostNet" Version="0.1.1" />
     <PackageReference Include="Microsoft.AspNetCore.App" />
-    <PackageReference Include="Microsoft.AspNetCore.Razor.Design" Version="2.2.0" PrivateAssets="All" />
+  </ItemGroup>
+
+
+  <ItemGroup>
+    <ProjectReference Include="..\CatBoostNet\CatBoostNet.csproj" />
   </ItemGroup>
 
 

--- a/catboost/dotnet/HeartDiseaseDemo/Startup.cs
+++ b/catboost/dotnet/HeartDiseaseDemo/Startup.cs
@@ -32,7 +32,7 @@ namespace HeartDiseaseDemo
             });
 
 
-            services.AddMvc().SetCompatibilityVersion(CompatibilityVersion.Version_2_2);
+            services.AddMvc().SetCompatibilityVersion(CompatibilityVersion.Version_2_1);
         }
 
         // This method gets called by the runtime. Use this method to configure the HTTP request pipeline.

--- a/catboost/dotnet/LibraryBuilder/LibraryBuilder.vcxproj
+++ b/catboost/dotnet/LibraryBuilder/LibraryBuilder.vcxproj
@@ -63,10 +63,10 @@
   </ImportGroup>
   <PropertyGroup Label="UserMacros" />
   <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Debug|x64'">
-    <NMakeBuildCommandLine>"$(SolutionDir)../../ya" make "$(SolutionDir)../../catboost/libs/model_interface" --output "$(ProjectDir)lib/$(Platform)/$(Configuration)" --verbose</NMakeBuildCommandLine>
+    <NMakeBuildCommandLine>call "$(VCInstallDir)\Auxiliary\Build\vcvars64.bat" &amp;&amp; "$(SolutionDir)../../ya" make "$(SolutionDir)../../catboost/libs/model_interface" --output "$(ProjectDir)lib/$(Platform)/$(Configuration)" --verbose</NMakeBuildCommandLine>
     <NMakePreprocessorDefinitions>_DEBUG;$(NMakePreprocessorDefinitions)</NMakePreprocessorDefinitions>
     <OutDir>$(ProjectDir)lib/$(Platform)/$(Configuration)</OutDir>
-    <NMakeReBuildCommandLine>"$(SolutionDir)../../ya" make "$(SolutionDir)../../catboost/libs/model_interface" --output "$(ProjectDir)lib/$(Platform)/$(Configuration)" --verbose --rebuild</NMakeReBuildCommandLine>
+    <NMakeReBuildCommandLine>call "$(VCInstallDir)\Auxiliary\Build\vcvars64.bat" &amp;&amp; "$(SolutionDir)../../ya" make "$(SolutionDir)../../catboost/libs/model_interface" --output "$(ProjectDir)lib/$(Platform)/$(Configuration)" --verbose --rebuild</NMakeReBuildCommandLine>
     <NMakeOutput>$(ProjectDir)lib/$(Platform)/$(Configuration)/catboost/libs/model_interface/catboostmodel.dl</NMakeOutput>
   </PropertyGroup>
   <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Debug|Win32'">
@@ -78,10 +78,10 @@
     <NMakePreprocessorDefinitions>WIN32;NDEBUG;$(NMakePreprocessorDefinitions)</NMakePreprocessorDefinitions>
   </PropertyGroup>
   <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Release|x64'">
-    <NMakeBuildCommandLine>"$(SolutionDir)../../ya" make "$(SolutionDir)../../catboost/libs/model_interface" --output "$(ProjectDir)lib/$(Platform)/$(Configuration)" --verbose</NMakeBuildCommandLine>
+    <NMakeBuildCommandLine>call "$(VCInstallDir)\Auxiliary\Build\vcvars64.bat" &amp;&amp; "$(SolutionDir)../../ya" make -r "$(SolutionDir)../../catboost/libs/model_interface" --output "$(ProjectDir)lib/$(Platform)/$(Configuration)" --verbose</NMakeBuildCommandLine>
     <NMakePreprocessorDefinitions>NDEBUG;$(NMakePreprocessorDefinitions)</NMakePreprocessorDefinitions>
     <OutDir>$(ProjectDir)lib/$(Platform)/$(Configuration)</OutDir>
-    <NMakeReBuildCommandLine>"$(SolutionDir)../../ya" make "$(SolutionDir)../../catboost/libs/model_interface" --output "$(ProjectDir)lib/$(Platform)/$(Configuration)" --verbose --rebuild</NMakeReBuildCommandLine>
+    <NMakeReBuildCommandLine>call "$(VCInstallDir)\Auxiliary\Build\vcvars64.bat" &amp;&amp; "$(SolutionDir)../../ya" make -r "$(SolutionDir)../../catboost/libs/model_interface" --output "$(ProjectDir)lib/$(Platform)/$(Configuration)" --verbose --rebuild</NMakeReBuildCommandLine>
     <NMakeOutput>$(ProjectDir)lib/$(Platform)/$(Configuration)/catboost/libs/model_interface/catboostmodel.dl</NMakeOutput>
   </PropertyGroup>
   <ItemDefinitionGroup>

--- a/catboost/dotnet/dotnet.sln
+++ b/catboost/dotnet/dotnet.sln
@@ -1,7 +1,7 @@
 ﻿
 Microsoft Visual Studio Solution File, Format Version 12.00
-# Visual Studio Version 16
-VisualStudioVersion = 16.0.28803.352
+# Visual Studio 15
+VisualStudioVersion = 15.0.28307.757
 MinimumVisualStudioVersion = 10.0.40219.1
 Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "CatBoostNet", "CatBoostNet\CatBoostNet.csproj", "{50D772D4-8573-4DAA-AB47-0DDAA5C271F3}"
 	ProjectSection(ProjectDependencies) = postProject

